.github/CONTRIBUTING.md

```diff
@@ -6,19 +6,86 @@ The following is a set of guidelines for contributing. These are mostly
+
+If this is your first time starting TruBudget, you should follow the [Developer Setup] for setting up the project.
+
+## How to ask for support
+
+If you need to ask for support feel free to reach out to us on
+[Gitter], check out the existing [documentation] or create a new [github issue]
+
+## How to contribute
+
+### Create an issue
+
+Feel free to help out if you noticed a bug or if you want to suggest a new feature by simply opening a new [github issue]
+
+### Open a Pull Request
+
+When working on a feature, you can open a PR in as soon as you push the first changes. Please make sure you follow these guidelines regarding PRs:
+
+- Make sure that the PR description clearly describes what you are working on
+- If aplicable, mention what issue will be closed with this pull request, by typing `Closes #issueNumber`
+- Describe how you are planning on implementing the soultion, maybe by creating a TODO list
+- The PR should be in draft mode if you're still making some changes. If it is ready to be reviewed then mark it as "Ready for review"
+
+### Don't forget to update the changelog!
+
+If you make changes that are relevant to the end user you should add an entry in the CHANGELOG.md file, which can be found in the root folder of the project. Before adding to the changelog, you should read these [guidelines]
+
+## Styleguides
+
+### Git Commits
+
+When writing commits you should consider the following guidelines:
+
+- Follow these [git commit guidelines]
+- Always include a prefix in the commit message with the abbreviation of the project you're working on (api, bc, doc, UI, e2e-test, excel-export, email-service)
+- When you're only changing the documentation you can include `[ci skip]` in the commit title
+
+### Git Branches
+
+When creating a new branch, you should consider the following guidelines regarding branch names:
+
+- Lead with the number of the issue you are working on
+- Add a short description of what the task is about
+- Use hyphens as separators
+
+### Git Workflow
 
-We use issues for everything, so can can ask questions as well as raise problems or
-suggestions by simply [creating a GitHub issue]. Also, feel free to reach out to us on
-[Gitter].
+To get an idea about the workflow used in our project you should read this [how to / git].
+So when working on a feature branch make sure to:
 
-[creating a github issue]: https://github.com/openkfw/TruBudget/issues/new
+1. Checkout the master branch and pull the recent changes
+2. Create a new feature branch respecting the guidelines mentioned above
+3. Try to keep the commits separate and respect the guidelines mentioned above. Don't squash the commits into a single one especially if you changed a lot of files
+4. Push to the remote repository and open a pull request respecting the guidelines mentioned above
+5. When your feature is ready make sure you have the latest changes by running `$ git pull --rebase origin master` on your feature branch
+6. Merge it into master
```

Comment:
@georgimld is this correct? We are not merging branches to master, we use pull request to do so.

Reply:
Yes, you're right. I meant merge it (the pull request) but it's not clear. I will rephrase it. Thanks!